core/cas-server-core-util/src/test/java/org/apereo/cas/util/RegexUtilsTests.java

```diff
@@ -0,0 +1,28 @@
+package org.apereo.cas.util;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests for {@link RegexUtils}
+ *
+ * @author David Rodriguez
+ * @since 5.0.0
+ */
+public class RegexUtilsTests {
+
+    @Test
+    public void verifyNotValidRegex() throws Exception {
```

Comment:
This is good. You also need to add test case for this scenario:

1. Create a registered service with an invalid pattern (serviceId)
2. Call matches on the service created.
3. Check the serviceId. The result should remain to be the same invalid serviceId set to 1.